.github/workflows/tests.yml

```
@@ -115,7 +115,7 @@ jobs:
       HOMEBREW_GITHUB_API_TOKEN: ${{secrets.GITHUB_TOKEN}}
     steps:
       - name: Check whether ARM tests will be forced
-        if: needs.tap_syntax.outputs.skip-arm-tests
+        if: needs.tap_syntax.outputs.skip-arm-tests == 'false'
```

```suggestion
        if: needs.tap_syntax.outputs.skip-arm-tests == 'true'
```

This needs to be flipped. We want to set `HOMEBREW_REQUIRE_BOTTLED_ARM` when the label _is not_ present (meaning when `skip-arm-tests` is true)

---

Just so I understand before commenting on this further, what does setting `HOMEBREW_REQUIRE_BOTTLED_ARM=1` do in practice? I see it's [used in test-bot](https://github.com/Homebrew/homebrew-test-bot/blob/bd7e2a22a215648d01ac02b257ffa72a0c470d4d/lib/tests/formulae.rb#L587) but I'm not familiar enough to understand what it's actually doing.

---

Yeah, the wording's a bit confusing here. First, all of this only applies to arm, the env variable is a no-op on anything else.

If `HOMEBREW_REQUIRE_BOTTLED_ARM` is set, `brew test-bot` will skip any formulae that aren't already bottled for arm. In other words, the variable is saying "an existing arm bottle is required for the test to run". That way, if a formula is not compatible with arm yet, we won't keep trying (and failing) to build bottles on every version bump. We set `HOMEBREW_REQUIRE_BOTTLED_ARM`, and the formula is skipped. However, when we're ready to bottle for arm, we want to be able to unset `HOMEBREW_REQUIRE_BOTTLED_ARM` so the tests _won't_ be skipped and can run.

When `CI-force-arm` is applied, we _don't_ want to require that formulae have ARM bottles to run their tests (i.e. we set the variable). When `CI-force-arm` is _not_ applies, we _do_ want to require ARM bottles to run tests (i.e. we don't set the variable).

So here's what we want (this table assumes a formula that isn't bottled for arm yet):

| `CI-force-arm` label? | `HOMEBREW_REQUIRE_BOTTLED_ARM` | ARM tests run? |
| --- | --- | --- |
| Yes | Unset | Yes |
| No | Set | No|

And with the `skip-arm-tests` intermediary:
| `CI-force-arm` label? | `skip-arm-tests` | `HOMEBREW_REQUIRE_BOTTLED_ARM` | ARM tests run? |
| --- | --- | --- | --- |
| Yes | `false` | Unset | Yes |
| No | `true` | Set | No |

Does that help? You can see how there are several negations in there which make it confusing.

---

I laughed when I saw your tables because I had created similar ones myself (glorified truth tables) to see how this all relates 😆

The missing piece for me was remembering that ARM builds happen separately, so the related `skip` in test-bot would only prevent ARM from being bottled. It didn't hit me until I started reading your explanation, so thanks for that.

I still feel like the `HOMEBREW_REQUIRE_BOTTLED_ARM` name doesn't explain the purpose as well as it could but maybe that's just me. I'm content to leave it alone until the next person comes along and we have to bring out the tables again, ahah.

---

Yeah, I love making tables for this stuff. Definitely helps to wrap my head around it.

In terms of the name, I agree it can definitely be changed. I almost suggested `HOMEBREW_SKIP_UNBOTTLED_FORMULAE_ARM_TESTS` but it's super long.

Since it's undocumented I don't think we need to worry about backwards compatibility, but it might be good to temporarily allow both names to be used in Homebrew/test-bot but I'd be fine just switching over, too.

---

If bottles will only ever be a formula thing (not cask) this could be shortened to `HOMEBREW_SKIP_UNBOTTLED_ARM_TESTS`. If that’s still too long, we could also drop `_TESTS` and even that may still be an improvement (though it’s more clear with it).

Naming is a pain and it never gets easier, ahah.